### DISCUSSION TITLE
Read only accounts

### DISCUSSION THREAD

**Discussion by sense4t** [paperless-ngx/paperless-ngx#252] [Feature Requests] [answered]
Hi,
i am successfully using paperless ng at home and now i want to implement it at my wife's business as well. In that case it would be critical to have an account that prohibits document deletions, as different team members should have read access to the paperless instance. Unfortunately the paperless user configuration does not seem to block deletions effectively.

Does anyone have an idea what's wrong here? Is this behavior a bug or have these restrictions not been implemented yet? Maybe i got something wrong here ?

Regards
Stefan

**Comment by alexthamm** [comment]
Hi @sense4t , paperless hasn't been designed as a multi user application. There was also this discussion in Jonas repository:
https://github.com/jonaswinkler/paperless-ng/discussions/52

I would love to use Paperless with others, but as er has written, for a clean implementation of access rights various places in Paperless would have to be considered.
If there is a concept for this, one could probably better assess whether the extension is implementable or possibly entails too many restrictions.

However, a read only user should actually be possible without huge modifications...

**Reply by sense4t** [reply]
Hi @alexthamm , 
i read the above discussion and  from my point of view it has been taken away into a manifold of requirements and ideas. 

To keep it simple i would propose to just lock any deletion of documents for everyone and to allow appending documents only. I would restrict this policy to the deletion of documents (nothing else, e.g. tags and correspondents can still be deleted).

Binding to and locking this policy for a specific user account maybe another step then.

Regards
Stefan

**Reply by alexthamm** [reply]
It's complicated ;) 
What you propose is more than a read only user. You want a user to be able to add docs but not to break anything. As soon as the user is allowed to change document metadata he could mess everything up. 
"Append only" is already possible by allowing the restricted user to add files into the consumer directory only. However, it would be up to the paperless user to provide the corresponding metadata for all the docs.

**Comment by sense4t** [comment]
I agree that it's crucial to have an overall (still simple) implementation concept for this, but at the moment the (automatically generated ?) django-konfiguration pretends to provide that functionality while it is not implemented. Another inconsistency related to that issue is to have user accounts at all ... 

Don't get me wrong: i really like (love :)) paperless-ng(x) as it provides even more funtionality than i need to manage my digital documents and it's really a great pice of work!

But sometimes (especially on late evenings) i would even restrict myself regarding deletions to not trash something by accident. 
From this perspective deletion prevention may not even be bound to having more than one user account necessarily but more a technical reliability issue (similar to the system setting lock on MacOS).

**Reply by sense4t** [reply]
btw: i DO make regular backups :)

**Reply by alexthamm** [reply]
Django provides everything needed to restrict users already but when you write your own UI instead of using the admin UI you have to explicitly consider the permissions. That wasn't necessary yet because paperless wasn't planned as a multi user system.

What is needed to support read only users?
The first step to support permission handling might be to restrict the API calls and adjust the Paperless UI to reflect these permissions. I never worked with Angular but these changes should not be too hard to implement. This would allow you to define a read only user.

More fine granular permissions require way more consideration that this. This includes users that can upload and modify their uploaded docs but not change existing docs. 
You also have to consider sensible information. Labels, correspondents and search suggestions could contain secrets you don't wan't to expose.

I am absolutely in favor of extending paperless to a multi user system. However, I have seen many problems with access rights in the enterprise search environment and therefore know that such a change is not trivial.

**Reply by sense4t** [reply]
I would been in favour of the first solution you described above, as it would fit to most of my use cases. 
Anything more would create an unnecessary amount of complexity at the moment.

**Comment by tooomm** [comment, marked as answer]
Linking https://github.com/paperless-ngx/paperless-ngx/discussions/295.
